components/builder-depot/src/server.rs

```diff
@@ -100,6 +102,21 @@ impl Key for TestableBroker {
     type Value = Self;
 }
 
+pub struct RoutedMessages(HashMap<TypeId, Vec<u8>>);
+
+impl RoutedMessages {
+    pub fn get<M: Routable>(&self) -> Result<M> {
+        let msg_type = &TypeId::of::<M>();
+        match self.0.get(msg_type) {
+            Some(msg) => {
+                Ok(parse_from_bytes::<M>(msg).expect(&format!("Unable to parse {:?} message",
```

Comment:
Here I do want a legitimate error for cases where I want to test that a route was purposely never called. Like ensuring that no data was ever sent to the database when user input was invalid.